### PR TITLE
Relax assumpitions in GridTools::partition_triangulation_zorder

### PR DESCRIPTION
**Pull request by peterrum**
I have added a single bool to be able to partition the space-filling curve up to 1 cell per process.

ping @tjhei @kronbichler

### CODE REVIEW

File: source/grid/grid_tools.inst.in [hunk 1]
```diff
@@ -18,7 +18,8 @@ for (X : TRIANGULATIONS; deal_II_dimension : DIMENSIONS;
   {
 #if deal_II_dimension <= deal_II_space_dimension
     namespace GridTools
-    \{
+
+    {
       template std::pair<
```
Comment (masterleinad) [inline source/grid/grid_tools.inst.in]: There seem to be a lot spurious changes in this file.

File: source/grid/grid_tools.inst.in [hunk 2]
```diff
@@ -366,86 +369,93 @@ for (deal_II_dimension : DIMENSIONS; deal_II_space_dimension : SPACE_DIMENSIONS)
...
-        unsigned int);
-#endif
-  }
+    for (deal_II_dimension : DIMENSIONS;
+         deal_II_space_dimension : SPACE_DIMENSIONS)
+      {
+#  if deal_II_space_dimension >= deal_II_dimension
+        namespace GridTools
+
+        {
```
Comment (masterleinad) [inline source/grid/grid_tools.inst.in]:
```suggestion
        \{
```

File: source/grid/grid_tools.inst.in (hunk 1 above)
Comment (peterrum) [reply source/grid/grid_tools.inst.in]: Sorry about this. I have fixed it!

File: include/deal.II/grid/grid_tools.h [hunk 3]
```diff
@@ -2006,15 +2006,22 @@ namespace GridTools
...
   /**
    * Generates a partitioning of the active cells making up the entire domain
-   * using the same partitioning scheme as in the p4est library. After calling
+   * using the same partitioning scheme as in the p4est library if the flag
+   * blocked is set to true (default behavior of this function). After calling
    * this function, the subdomain ids of all active cells will have values
    * between zero and @p n_partitions-1. You can access the subdomain id of a
    * cell by using <tt>cell-@>subdomain_id()</tt>.
+   *
+   * @note If the flag blocked is set to false, some assumptions of the p4est
```
Comment (tjhei) [inline include/deal.II/grid/grid_tools.h]: The only change is placing all terminal children of a single parent on a single process, right? Can you change the text here?

File: include/deal.II/grid/grid_tools.h [hunk 4]
```diff
@@ -2006,15 +2006,22 @@ namespace GridTools
 
   /**
    * Generates a partitioning of the active cells making up the entire domain
-   * using the same partitioning scheme as in the p4est library. After calling
+   * using the same partitioning scheme as in the p4est library if the flag
+   * blocked is set to true (default behavior of this function). After calling
```
Comment (tjhei) [inline include/deal.II/grid/grid_tools.h]:
```suggestion
   * @p blocked is set to true (default behavior of this function). After calling
```

File: include/deal.II/grid/grid_tools.h [hunk 5]
```diff
@@ -2006,15 +2006,22 @@ namespace GridTools
...
+   *       library are relaxed. E.g. p4est places all children of a cell on the
+   *       same processor if they are all active. By relaxing this, we can
+   *       can create partitions owning a single cell (also for refined meshes).
    */
   template <int dim, int spacedim>
   void
   partition_triangulation_zorder(const unsigned int            n_partitions,
-                                 Triangulation<dim, spacedim> &triangulation);
+                                 Triangulation<dim, spacedim> &triangulation,
+                                 const bool                    blocked = true);
```
Comment (tjhei) [inline include/deal.II/grid/grid_tools.h]: would you mind renaming this to ``group_siblings``, ``keep_siblings_on_one_process``, or something similar?

**Comment by peterrum** [comment]
@tjhei done!

**Pull request merged**